source/common/common/logger.cc

```diff
@@ -82,6 +82,14 @@ void DelegatingLogSink::set_formatter(std::unique_ptr<spdlog::formatter> formatt
   formatter_ = std::move(formatter);
 }
 
+spdlog::formatter* DelegatingLogSink::get_formatter() {
+  absl::MutexLock lock(&format_mutex_);
```

Comment:
Doesn't this introduce a potential deadlock if the signal handler runs while the thread is using the formatter already? Could you do something with `TryLock()` to make sure the signal handler doesn't get stuck?